### DISCUSSION TITLE
pandas_records column ordering

### DISCUSSION THREAD

**Discussion by pvardanis** [mlflow/mlflow#7963] [Q&A] [answered]
(also posted this on issues but maybe it's more appropriate to post it here as well)

Hi,

I've been using MLflow to serve models and currently been using the pandas records format. In the documentation it says:

"We do not recommend using this format because it is not guaranteed to preserve column ordering." Does this refer to the `pandas_df.to_dict()` command? To my knowledge since `Python 3.6` dictionaries are ordered, therefore this command should preserve column ordering.
"fine for vector rows, loses ordering for JSON records". I think it doesn't provide enough context. Can someone explain why this is the case?
The aforementioned documentation can be found [here](https://www.mlflow.org/docs/1.30.0/models.html#local-model-deployment). The `MLflow` version I'm using is `1.30.0`.

**Comment by WeichenXu123** [comment, marked as answer]
Replied here:
https://github.com/mlflow/mlflow/issues/7959#issuecomment-1455284443
